ddspipe_participants/src/cpp/writer/dds/CommonWriter.cpp

```diff
@@ -91,7 +97,7 @@ CommonWriter::CommonWriter(
     : BaseWriter(participant_id)
     , dds_participant_(participant)
     , dds_topic_(topic_entity)
-    , payload_pool_(payload_pool)
+    , payload_pool_(new core::PayloadPoolMediator(payload_pool))
```

Comment:
Leak spotted, I wouldn't use a raw pointer.

Reply:
No leak after all as you are setting a shared pointer. But, wouldn't it make more sense if it was a unique pointer?

Reply:
Fast-DDS's `create_datawriter` method requires it to be a shared_ptr.